osquery/utils/aws_util.cpp

```diff
@@ -477,4 +496,15 @@ Status appendLogTypeToJson(const std::string& log_type, std::string& log) {
   }
   return Status(0, "OK");
 }
+
+void setAWSProxy(Aws::Client::ClientConfiguration& config) {
+  if (FLAGS_aws_enable_proxy) {
```

Comment:
@fmanco if you have a suggestion about how to gate this from _outside_ of this function, I would be open to changing this logic. I looked at trying to add an optional argument to `makeAWSClient` that would support passing this in during client creation using this function, but it doesn't look like that's a great solution. thoughts?

Reply:
I'm fine with leaving this as is. But please update the documentation of this function bellow to make clear this function is a noop if `FLAGS_aws_enable_proxy` isn't set. Thanks!

Reply:
thanks @fmanco - I've updated the doc for the function to reflect your request